test/dates/ranges.jl

```diff
@@ -482,10 +482,10 @@ function testmonthranges2(n)
 end
 testmonthranges2(100000)
 
-@test length(Dates.Year(1):Dates.Year(10)) == 10
+@test length(Dates.Year(1):Dates.Year(1):Dates.Year(10)) == 10
```

Comment:
Why do we need to specify the step here? Can't we just define the following to get the old behaviour?
```julia
Base.colon{T<:Period}(start::T, stop::T) = StepRange(start, T(1), stop)
```